### PR TITLE
Add visibility-based splat filtering

### PR DESCRIPTION
**Pull request by slimbuck**
## Summary

Adds a new filter to remove the least visible/useful Gaussians from a scene based on their opacity and scale:

- New `--filter-visibility` (`-F`) CLI option
- Supports both count mode (`-F 10000`) and percentage mode (`-F 50%`)
- Visibility score = sigmoid(opacity) * exp(scale_0 + scale_1 + scale_2)
- Higher scores indicate more visible/important splats

## Usage

```bash
# Keep the top 50000 most visible splats
splat-transform input.ply --filter-visibility 50000 output.ply

# Keep the top 25% most visible splats  
splat-transform input.ply --filter-visibility 25% output.ply

# Combine with other operations
splat-transform input.ply -s 0.5 --filter-visibility 10000 -M output.ply
```

## Changes

- Add `sortByVisibility()` function in `src/lib/data-table/filter-visibility.ts`
- Add `FilterVisibility` process action type
- Add `-F, --filter-visibility` CLI option with count/percentage parsing
- Export new function and type from library index
- Add 23 tests covering visibility scoring, count/percent modes, and edge cases

### CODE REVIEW

File: src/cli/index.ts [hunk 1]
```diff
@@ -341,6 +368,8 @@ ACTIONS (can be repeated, in any order)
     -l, --lod              <n>              Specify the level of detail, n >= 0
     -m, --summary                           Print per-column statistics to stdout
     -M, --morton-order                      Reorder Gaussians by Morton code (Z-order curve)
+    -F, --filter-visibility <n|n%>          Keep the n most visible Gaussians (by opacity * volume)
+                                              Use n% to keep a percentage of Gaussians
```
Comment (willeastcott) [inline src/cli/index.ts]: This should be grouped with the other filters above.
